src/Avalonia.Base/Utilities/TypeUtilities.cs

```diff
@@ -13,7 +13,7 @@ namespace Avalonia.Utilities
     /// </summary>
     public static class TypeUtilities
     {
-        private static int[] Conversions =
+        private static readonly int[] Conversions =
```

Comment:
If we use `ReadOnlySpan<int>` here there won't be any array allocations, Roslyn will reference the data blob directly,
see https://github.com/dotnet/roslyn/pull/24621 and https://github.com/dotnet/corefx/pull/39678

Reply:
I decided to check this but it is unfortunately not true. If you check both ASM and IL for using `ROS<byte>` and `ROS<int>`  you will notice that only `byte` version actually does blob reference. If you check https://github.com/dotnet/corefx/pull/39678 in the code itself you will find that only byte is supported. https://github.com/dotnet/roslyn/blob/e782801c14884215d4abb9271ad0b360b0def592/src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs#L378-L381

[Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs7o64AcpsANwCGGGA1JI6AJRjCAJgHkAdrgCeAZV7DlAHjAALYVAB8zI1GmKNdALxmARMIBmDgHQBBAM5adACgBKAG4eTlC6fiFRcUJJGTklVU1tPWA1MTMAIXSYKxt7OmUYAHc6NLEAbQBdcPZKLga6AAY0OgB6NrpMiAhcOWU0cIbSVo6unr6dQfrGzhb2zu7e/unZzhGF8eWdcIBfEMpwyJExCSlZBRV1Xz1sZQwzAEl7vLszItK7jGraoa4m4CkJqkEGgsFAjZjJaTZR/OaApqI4Hg0HzMZMCxw5oIlG40adDTZMRYgEQ3Hg/FdHJ7A4NKi7IA)